maths/average_mode.py

```diff
@@ -1,36 +1,33 @@
-from typing import Any
-
-
-def mode(input_list: list) -> Any:  # Defining function "mode."
+def mode(input_list: list) -> list:  # Defining function "mode."
     """This function returns the mode(Mode as in the measures of
     central tendency) of the input data.
 
     The input list may contain any Datastructure or any Datatype.
 
     >>> input_list = [2, 3, 4, 5, 3, 4, 2, 5, 2, 2, 4, 2, 2, 2]
     >>> mode(input_list)
-    2
+    [2]
     >>> input_list = [3, 4, 5, 3, 4, 2, 5, 2, 2, 4, 4, 2, 2, 2]
     >>> mode(input_list)
-    2
-    >>> input_list = ["x", "y", "y", "z"]
+    [2]
+    >>> input_list = [3, 4, 5, 3, 4, 2, 5, 2, 2, 4, 4, 4, 2, 2, 4, 2]
     >>> mode(input_list)
-    'y'
-    >>> import statistics
-    >>> input_list = [2, 3, 4, 5, 3, 4, 2, 5, 2, 2, 4, 2, 2, 2]
-    >>> mode(input_list) == statistics.mode(input_list)
-    True
+    [2, 4]
     >>> input_list = ["x", "y", "y", "z"]
-    >>> mode(input_list) == statistics.mode(input_list)
-    True
+    >>> mode(input_list)
+    ['y']
+    >>> input_list = ["x", "x" , "y", "y", "z"]
+    >>> mode(input_list)
+    ['x', 'y']
     """
-    # Copying input_list to check with the index number later.
     result = list()  # Empty list to store the counts of elements in input_list
     for x in input_list:
         result.append(input_list.count(x))
     y = max(result)  # Gets the maximum value in the result list.

```

Comment:
This should have a check before this line so that if the result is an empty list, immediately return an empty list. Otherwise, `max` will raise an error saying `ValueError: max() arg is an empty sequence`.

```suggestion
    if not result:
        return []
    y = max(result)  # Gets the maximum value in the result list.
```